### PR TITLE
Remove no longer needed GMP, remove gcc from runner image

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ARG EXTRA_BUILD_APK_PACKAGES
 
 RUN apk add --no-cache --update \
     git tar build-base linux-headers autoconf automake libtool pkgconfig \
-    dbus-dev bzip2 bison flex gmp-dev cmake lz4 libsodium-dev openssl-dev \
```
Comment (jeffgrunewald) [inline Dockerfile:16]: this seems to cause issues with cargo in my testing:
```
sh: exec: line 1: cargo: not found
```

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ARG EXTRA_BUILD_APK_PACKAGES
...
+    dbus-dev bzip2 bison flex cmake lz4 libsodium-dev openssl-dev \
```
Comment (jeffgrunewald) [inline Dockerfile:16]:
```suggestion
    dbus-dev bzip2 bison flex gmp-dev cmake lz4 libsodium-dev openssl-dev \
```

File: Dockerfile [hunk 3]
```diff
@@ -42,7 +42,7 @@ FROM ${RUNNER_IMAGE} as runner
 ARG VERSION
 ARG EXTRA_RUNNER_APK_PACKAGES
 
-RUN apk add --no-cache --update ncurses dbus gmp libsodium gcc \
+RUN apk add --no-cache --update ncurses dbus libsodium \
```
Comment (Vagabond) [inline Dockerfile:45]:
```suggestion
RUN apk add --no-cache --update ncurses dbus libsodium libgcc \
```

File: Dockerfile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ARG BUILDER_IMAGE=erlang:23.3.4.6-alpine
-ARG RUNNER_IMAGE=erlang:23.3.4.6-alpine
+ARG RUNNER_IMAGE=alpine
```
Comment (jeffgrunewald) [inline Dockerfile:2]: This is just the default; you’ll need to make sure to update the same environment variable arg in the `.buildkite/scripts/make_image.sh`

Comment (Vagabond) [reply Dockerfile:2]: maybe you can do that?

Comment (Vagabond) [reply Dockerfile:2]: I did it.

**Review by jeffgrunewald — APPROVED** [review]

**Comment by lilacanaris2021** [comment]
Hi Jeff, I got this message too: 
sh: 1: exec: cargo: not found
===> Hook for compile failed!

Could you please explain what and where I have to change. 

Thanks for your support.  Regards Bert (I`m not an expert)

**Comment by jeffgrunewald** [comment]
@lilacanaris2021 I’m not able to reproduce that issue on either Mac or Linux but it implies you’re not successfully pulling down the rust tool chain in the step before compiling. Since we pull down the rustup from scratch it shouldn’t matter what your host platform is so long as it runs Docker. 

I would check the scroll back and see if there are any silent errors in retrieving the rust tool chain and ensure you’ve got proper dns resolution for that curl. It could just be transient network issue that resolves itself.

**Comment by lilacanaris2021** [comment]
Thanks a lot

**Pull request merged**